### ISSUE TITLE
uBlock origin plugin causes peer discovery to fail

### ISSUE THREAD

**Issue by chr15m** [webtorrent/webtorrent#1292]
This was not always the case but it seems that the uBlock Origin plugin is now blocking websocket connections to the default set of trackers.

Not sure if this should be considered a bug or not but these days quite a lot of people run ad-blocking plugins. [According to the Wikipedia page on uBlock Origin](https://en.wikipedia.org/wiki/UBlock_Origin):

>  the uBlock Origin Chrome extension has more than 10 million active users, and the Firefox version has more than 5 million active users.

**What version of WebTorrent?**

`v0.98.21`

**What browser and version? (if using WebTorrent in the browser)**

Firefox 58.0.1 on GNU/Linux.

**What did you expect to happen?**

Successful connection (screenshot is from Chrome without uBlock Origin running).

![chrome-webtorrent](https://user-images.githubusercontent.com/67130/36187337-84e05458-117f-11e8-8097-560074790de0.png)

**What actually happened?**

Only peer connected is self (screenshot is from Firefox on the same machine with uBlock Origin running).

![firefox-webtorrent](https://user-images.githubusercontent.com/67130/36187343-93ef6e70-117f-11e8-9d00-05a86471fe47.png)

Here is the set of domains uBlock Origin is blocking:

![ublock-origin](https://user-images.githubusercontent.com/67130/36187351-9ec28a8a-117f-11e8-8c8b-6c7555aaf984.png)

**Comment by DiegoRBaquero** [comment]
What's your config like? I don't see them blocked

**Comment by chr15m** [comment]
@DiegoRBaquero my config is the default, I haven't modified it. Is yours modified?

**Comment by feross** [comment]
Wow, I see this too now. The following rules are preventing connections to the default set of tracker servers:

```
.xyz^$third-party,websocket
||fastcast.nz^$third-party
||tracker.openwebtorrent.com^$third-party
```

I just opened an issue on the uBlock Origin tracker here: https://github.com/uBlockOrigin/uAssets/issues/1557 Will close this issue now.

**Comment by feross** [comment]
Both uBlock Origin and the EasyList maintainers have added exceptions for WebTorrent. ❤️

**Comment by DiegoRBaquero** [comment]
It seems they only unblocked in webtorrent.io (?) fastcast and openwebtorrent trackers won't work in btorrent.xyz

**Comment by feross** [comment]
The commit is here: https://github.com/easylist/easylist/commit/6cabf987a9e84d583c44fdc934b55f98e1448b7f

It's not 100% clear what they changed. I think they unblocked the fastcast and openwebtorrent trackers. But they continued blocking third-party and websocket connections to all .xyz domains, expect if you're on webtorrent.io.

So, maybe that means that btorrent.xyz users can still access tracker.btorrent.xyz because that's "first party" (on the same site). But it definitely seems that tracker.btorrent.xyz will be blocked from all domains that are not webtorrent.io and btorrent.xyz in uBlock Origin.

I think blocking all .xyz domains is quite broad. I wonder why they're doing that... I bet you could open another issue about it.
